ports/esp32s2/pcnt_handler.c

```diff
@@ -0,0 +1,67 @@
+/*
```

Comment:
Could you move this whole file to the ports/esp32s2/peripherals directory, and change the makefile correspondingly? We use that directory to organize additional driver functions and reservation systems for modules like this. You can see peripherals/rmt.c as an example.